File: 백준/Silver/1946. 신입 사원/신입 사원.py
# https://www.acmicpc.net/problem/1946
# 어떤 지원자 A의 성적이 다른 어떤 지원자 B의 성적에 비해 
# 서류 심사 결과와 면접 성적이 모두 떨어진다면 A는 결코 선발되지 않는다

import sys
input = sys.stdin.readline

T = int(input())
for _ in range(T):
    n = int(input())
    arr = [list(map(int, input().split())) for _ in range(n)]
    arr.sort()
    best_t1, best_t2 =arr[0]
    answer=[[best_t1,best_t2]]
    #print("arr",answer)
    
    for scores in arr:
        #print(scores, best_t2)
        if scores[1]<best_t2:
            answer.append(scores)
            best_t2 = scores[1]
               
    #print(answer)       
    print(len(answer))
        


"""
1
10
1 10
2 9
3 8
4 7
5 6
6 5
7 4
8 3
9 2
10 1

# Answer
10
"""

"""
## 반례 입력
1
5
1 5
2 3
3 4
4 2
5 1

## 정답
4
"""

File: 백준/Silver/1946. 신입 사원/신입 사원.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().split("\n")



const T = Number(input[0])
let line = 1
function solution(arr){
    let min_=1000001
    let count=0
    for(let i=0;i<arr.length;i++){
        const [x,y] = arr[i]
        if(min_>y){
            min_=y
            count+=1
        }
       
    }
    return count
    
}
for(let t=0;t<T;t++){
    const n = Number(input[line])
    let arr=[]
    for(let i=line+1;i<=line+n;i++){
        arr.push(input[i].split(" ").map(Number))
    }
    line+=n+1
    arr.sort((a,b)=>(a[0]-b[0])) // 서류 점수 낮은 사람이 가장 앞에 있음

    answer = solution(arr)

    console.log(answer)
}

// 다른 지원자의 비해 서류 더 떨어짐 && 면접 더 떨짐 => 탈락
// 내가 뒷 사람보다 서류 점수 낮음이 보장됨 
// 뒷사람보다 면접 등수라도 높아야 함 (숫자가 더 작아야 함)
// [ [ 5, 5 ], [ 4, 1 ], [ 3, 2 ], [ 2, 3 ], [ 1, 4 ] ]